HHS/Head-Start-TTADP
# [TTAHUB-1189] Fix deleting issues

## Description of change
Fixes a bug with goal info not being cleared out when the last goal is removed. This also explains the loading screen (attempting to fetch related info for the selected goal)

## How to test
To repro bug on main-ar-redesign
1) Add a goal to a new AR
2) Save draft
3) Save goal
4) Remove that goal, so that there are none left
You will see the form partially filled out, and there will be a loading screen

This PR fixes that issue

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1189


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
